### ISSUE TITLE
Generic definition with Generic Parameter Defaults in `<script setup lang="ts" generic="">` causes `"Parsing error: ';' expected.` after #220

### ISSUE THREAD

**Issue by danwithabox** [vuejs/vue-eslint-parser#221]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I'm using [eslint-plugin-vue](https://github.com/vuejs/eslint-plugin-vue).
- [X] I'm sure the problem is a parser problem. (If you are not sure, search for the issue in [eslint-plugin-vue](https://github.com/vuejs/eslint-plugin-vue) repo and open the issue in [eslint-plugin-vue](https://github.com/vuejs/eslint-plugin-vue) repo if there is no solution.
- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.

### What version of ESLint are you using?

8.56.0

### What version of `eslint-plugin-vue` and `vue-eslint-parser` are you using?

- vue-eslint-parser@9.4.1
- eslint-plugin-vue@9.20.1

### What did you do?

<details>
<summary>Configuration</summary>
I provide a stripped-down Vite project as the repro.

Issue is configuration-independent, any valid TS+Vue config will cause this.
</details>

```vue
<script setup lang="ts" generic="STitle extends string | undefined = undefined, SSubtitle extends string | undefined = undefined">
</script>
```

### What did you expect to happen?

No error on `Ln 1, Col 68`

![image](https://github.com/vuejs/vue-eslint-parser/assets/144792741/d0ec70e4-067a-4f62-bdaf-ef5aa5c19106)

### What actually happened?

Error on  `Ln 1, Col 68`

![image](https://github.com/vuejs/vue-eslint-parser/assets/144792741/c1a528fe-f889-4e89-926a-0a98cfbf7ce1)

### Link to Minimal Reproducible Example

https://github.com/danwithabox/issue_vue-eslint-parser_221

### Additional comments
[Generic Parameter Defaults](https://www.typescriptlang.org/docs/handbook/2/generics.html#generic-parameter-defaults) in `generic` attribute.

Works in `9.4.0`, fails in `9.4.1`

When searching for the issue, I found https://github.com/vuejs/vue-eslint-parser/issues/217 to be similar.
It seems to me, the fix for that, i.e. https://github.com/vuejs/vue-eslint-parser/pull/220 is probably the culprit.
